core/state_processor.go

```diff
@@ -232,3 +249,114 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, config, cfg)
 	return applyTransaction(msg, config, author, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
 }
+
+// keyValueMigrator is a helper struct that collects key-values from the base tree.
+// The walk is done in account order, so **we assume** the APIs hold this invariant. This is
+// useful to be smart about caching banderwagon.Points to make VKT key calculations faster.
+type keyValueMigrator struct {
+	currAddr      common.Address
+	currAddrPoint *verkle.Point
+
+	vktLeafData map[string]*verkle.BatchNewLeafNodeData
+}
```

Comment:
There shouldn't be much surprises here.
The TL;DR is that since we walk the tree by address, we keep the current address point cached and only update if a new account is detected. This makes sense considering how the caller uses this component.

We keep collecting stuff in the `vktLeafData` which we use at the end to batch create leaf nodes and insert into the tree. 

The method implementations shouldn't be surprising. They're doing something similar to what we did in the full tree conversion. Note that I'm sharing as soon as I finished writing the code, so might be something missing, so please double check.

I tried to use as many defined constants as possible to avoid magic indexes or sizes, etc.